apps/prairielearn/src/pages/instructorStudents/instructorStudents.tsx

```diff
@@ -0,0 +1,84 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { InsufficientCoursePermissionsCardPage } from '../../components/InsufficientCoursePermissionsCard.js';
+import { PageLayout } from '../../components/PageLayout.html.js';
+import { getCourseInstanceContext, getPageContext } from '../../lib/client/page-context.js';
+import { getCourseOwners } from '../../lib/course.js';
+import { Hydrate } from '../../lib/preact.js';
+import { getUrl } from '../../lib/url.js';
+
+import { InstructorStudents } from './instructorStudents.html.js';
+import { StudentQuerySchema, type StudentRow } from './instructorStudents.shared.js';
+
+const router = Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const pageContext = getPageContext(res.locals);
+    const { course_instance, course } = getCourseInstanceContext(res.locals, 'instructor');
+
+    const search = getUrl(req).search;
+
+    const hasPermission = pageContext.authz_data.has_course_instance_permission_view;
+    if (!hasPermission) {
+      const courseOwners = await getCourseOwners(course.id);
+      res.status(403).send(
+        InsufficientCoursePermissionsCardPage({
+          resLocals: res.locals,
+          courseOwners,
+          pageTitle: 'Students',
+          requiredPermissions: 'Instructor',
+        }),
+      );
+    }
+
+    const students: StudentRow[] = hasPermission
+      ? (
+          await queryRows(
+            sql.select_students,
+            {
+              course_instance_id: course_instance.id,
+            },
+            StudentQuerySchema,
+          )
+        ).map((student) => ({
+          ...student.user,
+          ...student.enrollment,
+        }))
```

Comment:
We generally don't want to do this. Like we talked about with Matt, we should be using objects that look like `{user: { ... }, enrollment: { ... }}`.

Reply:
Fixed in 1be0e4a6aeddafe6b0ea9fc3ee72f8e534369743.